sympy/combinatorics/polyhedron.py

```diff
@@ -822,6 +822,202 @@ def _string_to_perm(s):
+        Perm(0, 8, 15)(1, 12, 16)(2, 18, 10)(3, 19, 5)(4, 13, 14)(6, 7, 17)))
+
+icosahedron = Polyhedron(
+    Tuple(0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11),
+    Tuple(
+        Tuple(0, 1, 2),
+        Tuple(0, 2, 3),
+        Tuple(0, 3, 4),
+        Tuple(0, 4, 5),
+        Tuple(0, 1, 5),
+        Tuple(1, 6, 7),
+        Tuple(1, 2, 7),
+        Tuple(2, 7, 8),
+        Tuple(2, 3, 8),
+        Tuple(3, 8, 9),
+        Tuple(3, 4, 9),
+        Tuple(4, 9, 10),
+        Tuple(4, 5, 10),
+        Tuple(5, 6, 10),
+        Tuple(1, 5, 6),
+        Tuple(6, 7, 11),
+        Tuple(7, 8, 11),
+        Tuple(8, 9, 11),
+        Tuple(9, 10, 11),
+        Tuple(6, 10, 11)),
+    Tuple(
+        Perm(11)(1, 2, 3, 4, 5)(6, 7, 8, 9, 10),
+        Perm(0, 5, 6, 7, 2)(3, 4, 10, 11, 8),
+        Perm(0, 1, 7, 8, 3)(4, 5, 6, 11, 9),
+        Perm(0, 2, 8, 9, 4)(1, 7, 11, 10, 5),
+        Perm(0, 3, 9, 10, 5)(1, 2, 8, 11, 6),
+        Perm(0, 4, 10, 6, 1)(2, 3, 9, 11, 7),
+        Perm(0, 1)(2, 5)(3, 6)(4, 7)(8, 10)(9, 11),
+        Perm(0, 2)(1, 3)(4, 7)(5, 8)(6, 9)(10, 11),
+        Perm(0, 3)(1, 9)(2, 4)(5, 8)(6, 11)(7, 10),
+        Perm(0, 4)(1, 9)(2, 10)(3, 5)(6, 8)(7, 11),
+        Perm(0, 5)(1, 4)(2, 10)(3, 6)(7, 9)(8, 11),
+        Perm(0, 6)(1, 5)(2, 10)(3, 11)(4, 7)(8, 9),
+        Perm(0, 7)(1, 2)(3, 6)(4, 11)(5, 8)(9, 10),
+        Perm(0, 8)(1, 9)(2, 3)(4, 7)(5, 11)(6, 10),
+        Perm(0, 9)(1, 11)(2, 10)(3, 4)(5, 8)(6, 7),
+        Perm(0, 10)(1, 9)(2, 11)(3, 6)(4, 5)(7, 8),
+        Perm(0, 11)(1, 6)(2, 10)(3, 9)(4, 8)(5, 7),
+        Perm(0, 11)(1, 8)(2, 7)(3, 6)(4, 10)(5, 9),
+        Perm(0, 11)(1, 10)(2, 9)(3, 8)(4, 7)(5, 6),
+        Perm(0, 11)(1, 7)(2, 6)(3, 10)(4, 9)(5, 8),
+        Perm(0, 11)(1, 9)(2, 8)(3, 7)(4, 6)(5, 10),
+        Perm(0, 5, 1)(2, 4, 6)(3, 10, 7)(8, 9, 11),
+        Perm(0, 1, 2)(3, 5, 7)(4, 6, 8)(9, 10, 11),
+        Perm(0, 2, 3)(1, 8, 4)(5, 7, 9)(6, 11, 10),
+        Perm(0, 3, 4)(1, 8, 10)(2, 9, 5)(6, 7, 11),
+        Perm(0, 4, 5)(1, 3, 10)(2, 9, 6)(7, 8, 11),
+        Perm(0, 10, 7)(1, 5, 6)(2, 4, 11)(3, 9, 8),
+        Perm(0, 6, 8)(1, 7, 2)(3, 5, 11)(4, 10, 9),
+        Perm(0, 7, 9)(1, 11, 4)(2, 8, 3)(5, 6, 10),
+        Perm(0, 8, 10)(1, 7, 6)(2, 11, 5)(3, 9, 4),
+        Perm(0, 9, 6)(1, 3, 11)(2, 8, 7)(4, 10, 5)))
+
+tetrahedron_faces = [(0, 1, 2), (0, 2, 3), (0, 3, 1), (1, 2, 3)]
+
+cube_faces = [
+    (0, 1, 2, 3), (0, 1, 5, 4), (1, 2, 6, 5),
+    (2, 3, 7, 6), (0, 3, 7, 4), (4, 5, 6, 7)]
+
+octahedron_faces = [
+    (0, 1, 2), (0, 2, 3), (0, 3, 4), (0, 1, 4),
+    (1, 2, 5), (2, 3, 5), (3, 4, 5), (1, 4, 5)]
+
+dodecahedron_faces = [
+    (0, 1, 2, 3, 4), (0, 1, 6, 10, 5), (1, 2, 7, 11, 6),
+    (2, 3, 8, 12, 7), (3, 4, 9, 13, 8), (0, 4, 9, 14, 5),
+    (5, 10, 16, 15, 14), (6, 10, 16, 17, 11), (7, 11, 17, 18, 12),
+    (8, 12, 18, 19, 13), (9, 13, 19, 15, 14), (15, 16, 17, 18, 19)]
+
+icosahedron_faces = [
+    [0, 1, 2], [0, 2, 3], [0, 3, 4], [0, 4, 5], [0, 1, 5],
```

Comment:
nit: perhaps store as a list of tuples like above?

Reply:
How about creating the faces from the existing objects like `list(tuple(arg) for arg in icosahedron.faces)`?

Reply:
That gives the faces in a different order. Not sure if that matters or not.

Reply:
Done

Reply:
Usually, the reason why things should be systemically built is that some little things can change in the future, and it can be less prone to typos.

One other problem is that it can be possible to mutate the lists. Though I’m unsure that it can affect other references if used in library code.

Reply:
Do you mean that I should change it back to preserve the previous order?

Reply:
No, it doesn’t have to be reverted back.
Tetrahedron faces is still not derived, though.

Reply:
Oh yes. In fact changing it leads to a failure:
```
>       assert sorted(tetrahedron_faces) == sorted(tetrahedron_faces2)
E       assert [(0, 1, 2), (...3), (1, 2, 3)] == [(0, 1, 2), (0...1), (1, 2, 3)]
E         At index 1 diff: (0, 1, 3) != (0, 2, 3)
E         Full diff:
E         - [(0, 1, 2), (0, 1, 3), (0, 2, 3), (1, 2, 3)]
E         ?                 ^          ^  ^
E         + [(0, 1, 2), (0, 2, 3), (0, 3, 1), (1, 2, 3)]
E         ?                 ^          ^  ^

sympy/combinatorics/tests/test_polyhedron.py:102: AssertionError
```
I think these are equivalent though for triangular faces so I've changed the test to match this.